clients/TypeScript/packages/client/test/ChainSync.test.ts

```diff
@@ -46,24 +50,25 @@ describe('ChainSync', () => {
       await client.shutdown()
     })
 
-    it('accepts message handlers for roll back and roll forward messages', async () => {
+    it('requires message handlers to process roll back and roll forward messages, invoking the requestNext callback once ready for next message', async () => {
       const rollbackPoints: Point[] = []
       const blocks: Block[] = []
       const client = await createChainSyncClient({
-        rollBackward: ({ point }) => {
+        rollBackward: ({ point }, requestNext) => {
           rollbackPoints.push(point)
-          client.requestNext()
+          requestNext()
         },
-        rollForward: async ({ block }) => {
+        rollForward: async ({ block }, requestNext) => {
           if (blocks.length < 10) {
             blocks.push(block)
-            client.requestNext()
           }
+          requestNext()
         }
       }, {
-        connection
+        connection,
+        requestBuffer: 10
```

Comment:
Request buffer is perhaps a bit too vague to convey the right meaning? Especially in the context of networking where many things can be buffered. This refers to the number of requests in-flight, so perhaps simply `inFlight`?

Reply:
5ddae7e9054f3d8ebd2159915990ce8088609c03